prov/efa/src/rxr/rxr_ep.c

```diff
@@ -2156,35 +2143,39 @@ void rxr_ep_progress_internal(struct rxr_ep *ep)
 	/*
 	 * Send any queued ctrl packets.
 	 */
-	dlist_foreach_container_safe(&ep->rx_entry_queued_rnr_list,
+	dlist_foreach_container_safe(&ep->op_entry_queued_rnr_list,
 				     struct rxr_op_entry,
-				     rx_entry, queued_rnr_entry, tmp) {
-		peer = rxr_ep_get_peer(ep, rx_entry->addr);
+				     op_entry, queued_rnr_entry, tmp) {
+		peer = rxr_ep_get_peer(ep, op_entry->addr);
 		assert(peer);
 
 		if (peer->flags & RXR_PEER_IN_BACKOFF)
 			continue;
 
-		assert(rx_entry->rxr_flags & RXR_RX_ENTRY_QUEUED_RNR);
-		assert(!dlist_empty(&rx_entry->queued_pkts));
-		ret = rxr_ep_send_queued_pkts(ep, &rx_entry->queued_pkts);
+		assert(op_entry->rxr_flags & RXR_OP_ENTRY_QUEUED_RNR);
+		assert(!dlist_empty(&op_entry->queued_pkts));
+		ret = rxr_ep_send_queued_pkts(ep, &op_entry->queued_pkts);
 
 		if (ret == -FI_EAGAIN)
 			break;
 
 		if (OFI_UNLIKELY(ret)) {
-			rxr_cq_write_rx_error(ep, rx_entry, -ret, FI_EFA_ERR_PKT_SEND);
+			assert(op_entry->type == RXR_RX_ENTRY || op_entry->type == RXR_TX_ENTRY);
```

Comment:
Wonder why is the assertion necessary?

Reply:
There is another type of  operation that is `rxr_read_entry`, though `read_entry` will never encounter RNR, therefore the assertion.